### ISSUE TITLE
Vertical split size

### ISSUE THREAD

**Issue by nagy135** [nvim-neo-tree/neo-tree.nvim#467]
Hello, i m just curious if this behavior is intended, because it feels very unintuitive to me.

When you have vertical split (2 buffers side by side, 50/50 screen) and you open neo-tree (lets say, left side), what i would imagine happens is that those 2 buffers still are the same size, sharing 50/50 whats left with neo-tree opened. What actually happens is that both of these buffers _get pushed to right_ stealing real estate from right-most buffer.

So using ratios, we start with
50|50

Now open neo-tree on left.
Expected:
20|40|40
Reality:
20|50|30

This makes me use vim's equalize buffer size bind a lot (<kbd>c-w =</kbd>)

Does this bother anyone else and should this behave as I would expect?

**Comment by cseickel** [comment]
More often than not, if I start with equal size splits and I add a new split of a specific size, I would want it to subtract equal amounts from all windows. This is just not how vim works, however, and it's not always what I would want.

**Comment by MunifTanjim** [comment]
Working on https://github.com/MunifTanjim/nui.nvim/pull/165, I'm realizing that how the split size behaves is very hard to predict and control. There are multiple global options (`equalalways`, `eadirection`, `win{,min}{width,height}` to name a few) that affect split sizes. It's not always obvious how they behave and not always possible to control it effectively (or so it seems to me).

TLDR: I don't think it's something `neo-tree` can solve, because it depends on so many global options.

**Comment by nagy135** [comment]
okay, fair enough ...i will look into those options.
